plugin/README.md

```diff
@@ -0,0 +1,599 @@
+# Kubectl Plugin
+
+## Overview
+The kubectl plugin has been created in accordance with the instructions outlined in the [official documentation](https://kubernetes.io/docs/tasks/extend-kubectl/kubectl-plugins/).
+
+This plugin simplifies storage management for OpenEBS users who have deployed their clusters using the OpenEBS umbrella chart. It provides a unified interface for all OpenEBS storage engines like Mayastor, LocalPV-LVM, LocalPV-ZFS, and LocalPV-HostPath.
+
+This plugin leverages the existing ```kubectl mayastor``` codebase for Mayastor engine commands. For other engines (LocalPV-LVM, LocalPV-ZFS, and LocalPV-HostPath), new code has been developed, drawing inspiration from the current ```kubectl openebs``` plugin.
+
+While the original kubectl-openebs plugin only supported LocalPV engines, this plugin extends that support to include Mayastor and organizes commands by engine type for improved clarity and usability.
+
+## Namespace handling
+This plugin avoids the hardcoded "mayastor" namespace assumption used by the ```kubectl-mayastor``` plugin. Instead, if a namespace is not explicitly provided, the plugin defaults to the namespace defined in the default kubeconfig context. If ```--kube-config-path``` is specified, Then it picks namespace from the custom kubeconfig file if ```--namespace``` is not specified.
```

Comment:
```suggestion
This plugin avoids the hardcoded "mayastor" namespace assumption used by the ```kubectl-mayastor``` plugin. Instead, if a namespace is not explicitly provided, the plugin defaults to the namespace defined in the kubeconfig context. The kubeconfig path is configurable using `--kube-config-path` or the `KUBECONFIG` env. Namespace can be set explicitly using the `--namespace` option.
```

The kube-rs/kube library tries the KUBECONFIG env and the ~/.kube/config files for default cases.

Reply:
That's right, the default kubeconfig path is first read from the env, and if that's empty or unset, then the default kubeconfig path $HOME/.kube/config path is used.

Please follow this call stack:
1. https://github.com/openebs/openebs/blob/develop/plugin/src/main.rs#L30
2. https://github.com/openebs/mayastor-control-plane/blob/develop/k8s/proxy/src/lib.rs#L39
3. https://github.com/openebs/mayastor-control-plane/blob/develop/k8s/proxy/src/lib.rs#L28C31-L28C46
4. https://github.com/kube-rs/kube/blob/main/kube-client/src/config/file_loader.rs#L29
5. https://github.com/kube-rs/kube/blob/main/kube-client/src/config/file_config.rs#L412